src/gui/qgsmaptoolcapture.cpp

```diff
@@ -63,6 +72,8 @@ QgsMapToolCapture::~QgsMapToolCapture()
     mValidator->deleteLater();
     mValidator = nullptr;
   }
+  mExtraSnapLayer->deleteLater();
```

Comment:
```suggestion
  mCanvas->snappingUtils()->removeExtraSnapLayer( mExtraSnapLayer );
  mExtraSnapLayer->deleteLater();
```

Or is this guaranteed to be taken care of elsewhere?